experimental/apps-mcp/cmd/mcp.go

```diff
@@ -0,0 +1,91 @@
+package mcp
+
+import (
+	"github.com/databricks/cli/cmd/root"
+	mcplib "github.com/databricks/cli/experimental/apps-mcp/lib"
+	"github.com/databricks/cli/experimental/apps-mcp/lib/server"
+	"github.com/databricks/cli/libs/cmdctx"
+	"github.com/databricks/cli/libs/log"
+	"github.com/spf13/cobra"
+)
+
+func NewMcpCmd() *cobra.Command {
+	var warehouseID string
+	var allowDeployment bool
+	var withWorkspaceTools bool
+	var dockerImage string
+	var useDagger bool
+
+	cmd := &cobra.Command{
+		Use:   "apps-mcp",
+		Short: "Model Context Protocol server for AI agents",
+		Long: `Start and manage an MCP server that provides AI agents with tools to interact with Databricks.
+
+The MCP server exposes the following capabilities:
+- Databricks integration (query catalogs, schemas, tables, execute SQL)
+- Project scaffolding (generate full-stack TypeScript applications)
+- Sandboxed execution (isolated file/command execution)
+
+The server communicates via stdio using the Model Context Protocol.`,
+		Example: `  # Start MCP server with required warehouse
+  databricks experimental apps-mcp --warehouse-id abc123
```

Comment:
Non-blocking: we should apply the "default warehouse" logic from https://github.com/databricks/cli/pull/3907/files?w=1#diff-5af5f7c0874402273882f7e0c97c9bc7a0471ad92771d90ef23c415da090f09aR74 here. That same logic, discussed with the default warehouse team, is also applied in many other client-side tools. We'll get a proper API for it in the future.